src/h5web/visualizations/line/DataCurve.tsx

```diff
@@ -19,21 +19,29 @@ function DataCurve(props: Props): JSX.Element {
   const { values, color = DEFAULT_COLOR } = props;
 
   const { abscissaInfo, ordinateInfo } = useContext(AxisSystemContext);
-  const { size } = useThree();
+  const { camera, size } = useThree();
   const { width, height } = size;
 
   const dataGeometry = useMemo(() => {
     const abscissaScale = getAxisScale(abscissaInfo, width);
     const ordinateScale = getAxisScale(ordinateInfo, height);
 
     const points = values.map(
-      (val, index) => new Vector3(abscissaScale(index), ordinateScale(val), 0)
+      (val, index) =>
+        new Vector3(
+          abscissaScale(index),
+          // This is to avoid a three.js warning when ordinateScale(val) is Infinity
+          Number.isFinite(ordinateScale(val)) ? ordinateScale(val) : 0,
+          // Move NaN/Infinity out of the camera FOV (negative val for logScale).
+          // This allows to have only curve segments for the positive values
+          Number.isFinite(ordinateScale(val)) ? 0 : camera.far
+        )
```

Comment:
While a bit clunky, this allows to move out of the FOV the data points corresponding to negative values **and** the lines connecting to them when displaying the lines.
Simply removing negative values from the data would create a curve where all points are connected which is not wanted.

Reply:
Very clever